frontend/src/app/modules/user-profile/user-profile-page/user-profile.component.ts

```diff
@@ -0,0 +1,215 @@
+            ],
+            repeatPassword: ['', [Validators.required, ValidationsFn.matchValues('newPassword')]],
+        });
+
+        this.passwordForm.controls['newPassword'].valueChanges.subscribe(() => {
+            this.passwordForm.controls['repeatPassword'].updateValueAndValidity();
+        });
+    }
+
+    private initNotificationsValue() {
+        this.squirrelNotification = this.currentUser.squirrelNotification;
+        this.emailNotification = this.currentUser.emailNotification;
+    }
+
+    public updateUserNames() {
+        if (!this.userNamesForm.valid) {
+            this.notificationService.error('Update Names Form is invalid');
+
+            return;
+        }
+
+        this.spinner.show();
+        const userData: UpdateUserNamesDto = {
+            userName: this.userNamesForm.value.userName,
+            firstName: this.userNamesForm.value.firstName,
+            lastName: this.userNamesForm.value.lastName,
+        };
+
+        const userSubscription = this.userService.updateUserNames(userData);
+
+        userSubscription.pipe(takeUntil(this.unsubscribe$)).subscribe(
+            (user) => {
+                this.currentUser = user;
+                this.spinner.hide();
+                this.notificationService.info('Names successfully updated');
+                this.initUserNamesForm();
+            },
+            (error) => {
+                this.spinner.hide();
+                this.notificationService.error(error.message);
+            },
+        );
+    }
+
+    public updateUserPassword() {
+        if (
+            !(this.passwordForm.valid && this.passwordForm.value.newPassword === this.passwordForm.value.repeatPassword)
+        ) {
+            this.notificationService.error('Update Names Form is invalid');
+
+            return;
+        }
+
+        this.spinner.show();
+        const userData: UpdateUserPasswordDto = {
+            currentPassword: this.passwordForm.value.currentPassword,
+            newPassword: this.passwordForm.value.newPassword,
+        };
+
+        const userSubscription = this.userService.updateUserPassword(userData);
+
+        userSubscription.pipe(takeUntil(this.unsubscribe$)).subscribe(
+            () => {
+                this.spinner.hide();
+                this.notificationService.info('Password successfully updated');
+                this.initChangePasswordForm();
+            },
+            (error) => {
+                this.spinner.hide();
+                this.notificationService.error(error.message);
+            },
+        );
+    }
+
+    public updateUserNotifications() {
+        this.spinner.show();
+        const userData: UpdateUserNotificationsDto = {
+            squirrelNotification: this.squirrelNotification,
+            emailNotification: this.emailNotification,
+        };
+
+        const userSubscription = this.userService.updateUserNotifications(userData);
+
+        userSubscription.pipe(takeUntil(this.unsubscribe$)).subscribe(
```

Comment:
no need for the `userSubscription` variable. Remove such subscription variables everywhere